src/app/application/services/data-source/index.ts

```diff
@@ -84,11 +83,9 @@ export const create: Creator = (src, router) => {
     // TODO(jc) Once we remove all the source.closes in the sources.ts files the
     // second argument here can go
     const init = route.route(params, { close: () => {} })
-    if (init instanceof CallableEventSource) {
-      return init
-    } else {
-      return source(() => Promise.resolve(init))
-    }
+    const eventSource = init instanceof CallableEventSource ? init : source(() => Promise.resolve(init))
+    eventSource.url = src
+    return eventSource
```

Comment:
These are changes I made whilst exactly tracing/loggin/debugging how things were working, I left them in as they are kinda useful. Little caveat here is that `url` should really be `readonly`, but I didn't want to do too much refactoring here to make this a little more protected. As it stands now sometimes its handy to be able to see the DataSources `src`/`uri` easily.